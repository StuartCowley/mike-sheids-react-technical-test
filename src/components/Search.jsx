import React, { useState } from "react";
import "../styles/search.css";

const Search = () => {
  return (
    <>
      <input
        className="search-input"
        placeholder="enter your search"
        type="text"
      />
    </>
  );
};

export default Search;
