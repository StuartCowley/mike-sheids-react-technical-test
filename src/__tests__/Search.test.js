import React from "react";
import { render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe("Search component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Search />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays a text input", () => {
    render(<Search />);

    expect(
      screen.getByPlaceholderText(/enter your search/i)
    ).toBeInTheDocument();
  });
});
