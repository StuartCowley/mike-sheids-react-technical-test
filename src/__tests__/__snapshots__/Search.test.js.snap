// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component matches snapshot 1`] = `
<DocumentFragment>
  <input
    class="search-input"
    placeholder="enter your search"
    type="text"
  />
</DocumentFragment>
`;
